src/builder/edit_current_application_info.rs

```diff
@@ -0,0 +1,182 @@
+use std::borrow::Cow;
+use std::collections::HashMap;
+
+#[cfg(feature = "http")]
+use crate::http::Http;
+use crate::model::prelude::*;
+
+/// A builder for editing [`CurrentApplicationInfo`] i.e the current Application's information.
+///
+/// The fields are optional, and only the ones explicitly set will be updated.
+#[derive(Clone, Debug, Default, Serialize)]
+#[must_use]
+pub struct EditCurrentApplicationInfo<'a> {
+    /// Default custom authorization URL for the app, if enabled.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    custom_install_url: Option<Cow<'a, str>>,
+
+    /// Description of the app.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    description: Option<Cow<'a, str>>,
+
+    /// Role connection verification URL for the app.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    role_connections_verification_url: Option<Cow<'a, str>>,
+
+    /// Settings for the app's default in-app authorization link, if enabled.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    install_params: Option<InstallParams>,
+
+    /// Default scopes and permissions for each supported installation context.
+    /// Value for each key is an integration type configuration object.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    integration_types_config: Option<HashMap<InstallationContext, InstallationContextConfig>>,
+
+    /// App's public flags.
+    ///
+    /// Only limited intent flags (GATEWAY_PRESENCE_LIMITED, GATEWAY_GUILD_MEMBERS_LIMITED,
+    /// and GATEWAY_MESSAGE_CONTENT_LIMITED) can be updated via the API.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    flags: Option<ApplicationFlags>,
+
+    /// Icon for the app
+    #[serde(skip_serializing_if = "Option::is_none")]
+    icon: Option<ImageHash>,
+
+    /// Default rich presence invite cover image for the app.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    cover_image: Option<ImageHash>,
+
+    /// Interactions endpoint URL for the app.
+    ///
+    /// To update an Interactions endpoint URL via the API, the URL must be valid according
+    /// to the [Receiving an Interaction]
+    /// (https://discord.com/developers/docs/interactions/receiving-and-responding#receiving-an-interaction) documentation.
```

Comment:
These docs are private and therefore unnecessary. I'd recommend instead incorporating them into the docs for the methods themselves.

Reply:
Ah, of course. Fixed it now.